src/main/java/com/mraof/minestuck/world/gen/feature/StructureBlockRegistryProcessor.java

```diff
@@ -1,47 +1,49 @@
 package com.mraof.minestuck.world.gen.feature;
 
 import com.mojang.serialization.Codec;
-import com.mojang.serialization.Decoder;
-import com.mojang.serialization.Encoder;
 import com.mraof.minestuck.world.gen.structure.blocks.StructureBlockRegistry;
 import net.minecraft.MethodsReturnNonnullByDefault;
 import net.minecraft.core.BlockPos;
-import net.minecraft.world.level.LevelReader;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.level.ServerLevelAccessor;
 import net.minecraft.world.level.block.state.BlockState;
-import net.minecraft.world.level.levelgen.feature.FeaturePlaceContext;
-import net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings;
-import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor;
-import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorType;
-import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
+import net.minecraft.world.level.chunk.ChunkGenerator;
+import net.minecraft.world.level.levelgen.structure.templatesystem.*;
 
-import javax.annotation.Nullable;
 import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
 
 @ParametersAreNonnullByDefault
 @MethodsReturnNonnullByDefault
 public class StructureBlockRegistryProcessor extends StructureProcessor
 {
-	public static final Codec<StructureBlockRegistryProcessor> CODEC = Codec.of(Encoder.error("StructureBlockRegistryProcessor is not serializable."), Decoder.error("StructureBlockRegistryProcessor is not serializable."));
-	private final StructureBlockRegistry blocks;
+	public static final Codec<StructureBlockRegistryProcessor> CODEC = Codec.unit(() -> StructureBlockRegistryProcessor.INSTANCE);
+	public static final StructureBlockRegistryProcessor INSTANCE = new StructureBlockRegistryProcessor();
 	
-	public StructureBlockRegistryProcessor(StructureBlockRegistry blocks)
+	private StructureBlockRegistryProcessor()
 	{
-		this.blocks = blocks;
 	}
 	
-	public static StructureBlockRegistryProcessor from(FeaturePlaceContext<?> context)
-	{
-		return new StructureBlockRegistryProcessor(StructureBlockRegistry.getOrDefault(context.chunkGenerator()));
-	}
-	
-	@Nullable
-	@Override	//TODO figure out blockpos difference
```

Comment:
Should this TODO be included in the new code?

Reply:
From what I recall and since the two position fields were never used in this processor, that TODO was just about figuring out more meaningful names for `blockPos` and `blockPos2`. I think that's solved in the new code, even if I still don't fully know what the technical difference between the two are. You can leave it out.

Reply:
Based on your explanation, I would theorize that the second block pos is used by the gravity processor. Regardless, I will leave it out!